llvm/lib/Target/RISCV/RISCVFrameLowering.cpp

```diff
@@ -1418,6 +1605,17 @@ void RISCVFrameLowering::determineCalleeSaves(MachineFunction &MF,
     SavedRegs.set(RISCV::X27);
 }
 
+void RISCVFrameLowering::determineCalleeSaves(MachineFunction &MF,
+                                              BitVector &SavedRegs,
+                                              RegScavenger *RS) const {
+  const auto &ST = MF.getSubtarget<RISCVSubtarget>();
+  determineMustCalleeSaves(MF, SavedRegs);
```

Comment:
It looks like we're doing additional logic that did not exist before in the case that `!ST.doCSRSavesInRA()`. All the logic before

```
if (hasFP(MF)) {
    SavedRegs.set(RISCV::X1);
    SavedRegs.set(RISCV::X8);
  }
```

in `determineMustCalleeSaves` is new. Is all of that additional logic desired?